### PR TITLE
Stringify errors in test app

### PR DESCRIPTION
**Pull request by jekloudaMSFT**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

> Some errors are currently not being stringified in the test app, which leads to cases where the output for an API will be "Error: [object object]". This makes debugging more difficult since it is not immediately apparent exactly what error the API threw. This PR just calls JSON.stringify on errors that are not already strings.

### Main changes in the PR:

1. Call JSON.stringify on test app errors that are not already strings

### Validation performed:

1. Tested the three APIs that are currently hardcoded to validate "Error: [object Object]" in the e2e definitions

### Unit Tests added:

> No, test app changes only

### End-to-end tests added:

> No

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

No

### Screenshots:

> Remove this section if n/a

| Before     | After      |
| ---------- | ---------- |
| image | image |

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -110,17 +110,13 @@ jobs:
     displayName: 'E2E Test - CDN (only runs on release builds)'
     # This test only runs after deployment from a release branch and the new CDN version has been deployed
     # This check will run on the PR to merge the release branch back into main
-    condition: and(
-      eq(variables['Build.Reason'], 'PullRequest'),
-      startsWith(variables['System.PullRequest.SourceBranch'], 'release/'),
-      eq(variables['System.PullRequest.TargetBranch'], 'main')
```
Comment (erinha) [inline azure-pipelines.yml:116]: Why are these conditions being removed?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -26,14 +26,14 @@ jobs:
     steps:
       - template: tools/yaml-templates/build-test-publish.yml
 
-  - job: E2ETest1
+  - job: E2ETest11
```
Comment (erinha) [inline azure-pipelines.yml:29]: Intentional change?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -26,14 +26,14 @@ jobs:
...
     displayName: 'E2E Test - Perf'
     pool:
       vmImage: 'ubuntu-latest'
     steps:
       - template: tools/yaml-templates/build-app-host.yml
         parameters:
-          appHostGitPath: git://$(AppHostingSdkGitPath)@$(AppHostingSdkGitRef)
+          appHostGitPath: git://$(AppHostingSdkGitPath)@kx/fix_pull_data_bug
```
Comment (erinha) [inline azure-pipelines.yml:36]: Please make sure not to check these changes in.

File: azure-pipelines.yml (hunk 2 above)
Comment (KangxuanYe) [reply azure-pipelines.yml:29]: Yes, for triggering PR. Just for testing purpose

File: azure-pipelines.yml (hunk 1 above)
Comment (KangxuanYe) [reply azure-pipelines.yml:116]: Removed for testing E2E test on release.

File: azure-pipelines.yml (hunk 3 above)
Comment (KangxuanYe) [reply azure-pipelines.yml:36]: Yeah definitely. It will be switched back once testing is done.

File: azure-pipelines.yml (hunk 1 above)
Comment (erinha) [reply azure-pipelines.yml:116]: So not intended for check-in?

File: apps/teams-test-app/src/components/utils/ApiWithTextInput.tsx [hunk 4]
```diff
@@ -55,7 +55,11 @@ export const ApiWithTextInput = <T extends unknown>(props: ApiWithTextInputProps
         }
       }
     } catch (err) {
-      setResult('Error: ' + err);
+      let error = `${err}`;
+      if (error === '[object Object]') {
+        error = JSON.stringify(err);
+      }
+      setResult('Error: ' + error);
```
Comment (erinha) [inline apps/teams-test-app/src/components/utils/ApiWithTextInput.tsx:62]: This code is duplicated in ApiWithoutInput.tsx. Can we make this a helper function that is called in both places?

File: azure-pipelines.yml [hunk 5]
```diff
@@ -55,9 +55,9 @@ jobs:
     steps:
       - template: tools/yaml-templates/build-app-host.yml
         parameters:
-          appHostGitPath: git://$(AppHostingSdkGitPath2)@$(AppHostingSdkGitRef)
+          appHostGitPath: git://$(AppHostingSdkGitPath2)@kx/comment_error_object_object_case
 
-      - bash: 'node tools/cli/runAppsWithE2ETests.js --useDataFromLocal=true'
+      - bash: 'node tools/cli/runAppsWithE2ETests.js  --useDataFromLocal=true'
```
Comment (erinha) [inline azure-pipelines.yml:60]: Is this intended?

**Review by KangxuanYe — DISMISSED** [review]

**Review by KangxuanYe — APPROVED** [review]

**Review by TrevorJoelHarris — APPROVED** [review]

**Pull request merged**